spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/core/subscriber/PubSubSubscriberTemplate.java

```diff
@@ -117,13 +118,25 @@ public Subscriber subscribe(String subscription,
 	@Override
 	public List<AcknowledgeablePubsubMessage> pull(String subscription, Integer maxMessages,
 			Boolean returnImmediately) {
+		Assert.hasText(subscription, "The subscription can't be null or empty.");
+
+		if (maxMessages != null) {
```

Comment:
I believe we can combine both conditions into one `Assert.isTrue()` call

Reply:
I believe they are better separated so that code can provide better explanation of the issue. (`subscription` issue vs `maxMessages` range issue.)